react-components/src/data/edit/EditComments.js

```diff
@@ -0,0 +1,207 @@
+
+        }
+        this.setState({commentList : temp});
+    };
+
+    handleSortOwner = () => {
+        console.log("Sort User");
+        let swapped;
+        let aux = 0;
+        let temp = this.state.commentList;
+        swapped = true;
+        while(swapped){
+            swapped = false;
+            for(let i = 1; i < temp.length; i++){
+                if(temp[i - 1].owner.charAt(0) > temp[i].owner.charAt(0)){
+                    swapped = true;
+                    aux = temp[i - 1];
+                    temp[i - 1] = temp[i];
+                    temp[i] = aux;
+                }
+            }
+
+        }
+        this.setState({commentList : temp});
+    };
+
+    handleChange = name => event => {
+        this.setState({
+            [name]: event.target.value,
+        });
+    };
+
+    createComment = () => {
+      const text = document.getElementById("addCommentTextField").value;
+      console.log(text);
+    };
+
+
+    render(){
+        const {classes} = this.props;
+        let commentItems = this.state.commentList ? this.state.commentList.map((comment, index) =>
+            <Comment message={comment.message} id={comment.id} retracted={comment.retracted}
+                    date={comment.date} owner={comment.owner}/>
+    ) : [];
+        return(
+            <div className={classes.root}>
+                <Dialog
+                    open={this.state.open}
+                    onClose={this.handleClose}
+                    id={"edit-comments-dialog"}
+                    className={classes.main}
+                >
+                    <DEDialogHeader
+                        id={"edit-comments-dialog-title"}
+                        heading={"Comments"}
+                        onClose={this.handleClose}
+                    >
+
+                    </DEDialogHeader>
+                    <hr></hr>
+
+                    <Button variant="contained" color="secondary" className={classes.deleteButton}
+
+                    >
+                        Retract
+                        <DeleteIcon />
+                    </Button>
+
+                        <label id={"editCommentsDropDownLabel"} className={classes.dropDownLabel}>  Comments</label>
+                        <DotMenu
+                            handleSortMostRecent={this.handleSortMostRecent}
+                            handleSortLeastRecent={this.handleSortLeastRecent}
+                            handleSortOwner={this.handleSortOwner}
+                            className={this.props.classes.dropDownDots}
+                        />
+
+                    <DialogContent id="editCommentsCommentList" className={classes.dContent}>
+                        <List component="nav">
+                            {commentItems}
+                        </List>
+                    </DialogContent>
+                    <TextField
+                        value={this.state.multiline}
+                        onChange={this.handleChange('multiline')}
```

Comment:
this will immediately invoke the function `handleChange`. If you want to invoke a function onChange event, the you need to do something like `onChange={(parameterTopass)=>this.handleChange(parameterTopass)}`